pkg/cmd/template/data_values_flags.go

```diff
@@ -149,34 +152,43 @@ func (s *DataValuesFlags) file(fullPath string, strict bool) ([]*datavalues.Enve
 		return nil, err
 	}
 
-	contents, err := s.readFile(path)
-	if err != nil {
-		return nil, fmt.Errorf("Reading file '%s': %s", path, err)
-	}
-
-	docSetOpts := yamlmeta.DocSetOpts{
-		AssociatedName: path,
-		Strict:         strict,
-	}
-
-	docSet, err := yamlmeta.NewDocumentSetFromBytes(contents, docSetOpts)
+	dvFiles, err := s.asFiles(path)
 	if err != nil {
-		return nil, fmt.Errorf("Unmarshaling YAML data values file '%s': %s", path, err)
+		return nil, fmt.Errorf("Find files '%s': %s", path, err)
 	}
 
 	var result []*datavalues.Envelope
+	for _, dvFile := range dvFiles {
+		// skip non-YAML files that are contained within the specified directory
+		if strings.Contains(dvFile.RelativePath(), "/") && !(dvFile.Type() == files.TypeYAML) {
```

Comment:
> strings.Contains(dvFile.RelativePath(), "/")

whats this bit about?

Reply:
If the user supplies a non-YAML file directly as the parameter for the flag, we don't want to ignore it. It's only _implied_ files (i.e. contained within a subdirectory) that we want to ignore.

Reply:
... although, I had intended to check if `RelativePath()` was returning a normalized path or if we needed to grab the OS-specific path separator token, here.

I'll do that.

Reply:
_(discussed in [Slack :thread: in #carvel](https://kubernetes.slack.com/archives/CH8KCCKA5/p1653056007351229?thread_ts=1653053084.082349&cid=CH8KCCKA5))_

Reply:
Added `File.IsImplied()` and make check OS-aware.